models/issue_comment.go

```diff
@@ -403,16 +403,23 @@ func (c *Comment) mailParticipants(e Engine, opType ActionType, issue *Issue) (e
 		return fmt.Errorf("UpdateIssueMentions [%d]: %v", c.IssueID, err)
 	}
 
-	content := c.Content
+	if len(c.Content) > 0 {
+		if err = mailIssueCommentToParticipants(e, issue, c.Poster, c.Content, c, mentions); err != nil {
+			log.Error("mailIssueCommentToParticipants: %v", err)
+		}
```

Comment:
return?

Reply:
No, sending two mails is intended. One for each action (comment and close/reopen).

Reply:
If it's just a comment, the `switch` below will not hit a case and only 1 mail is sent.

Reply:
One action sending two mails is annoying.

Reply:
I don't think so. It's not one action after all if a user clicks "comment and close". Also it does match GitHub's behaviour. They also send two mails on "comment and close".

Reply:
Example of GitHub sending two mails for "comment and close":

![image](https://user-images.githubusercontent.com/115237/59181957-c9cc9b80-8b68-11e9-87d2-3c9f2cfe6e74.png)